src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/ChatWithCustomData/ChatWithCustomData.Web/Program.cs

```diff
@@ -134,3 +145,73 @@ await DataIngestor.IngestDataAsync(
     new PDFDirectorySource(Path.Combine(builder.Environment.ContentRootPath, "Data")));
 
 app.Run();
+
+#if (IsOllama)
+async Task ValidatePrerequisitesAsync(IConfiguration configuration)
+#else
+void ValidatePrerequisites(IConfiguration configuration)
+#endif
+{
+#if (IsOllama)
+    var client = new OllamaApiClient(new Uri("http://localhost:11434"));
```

Comment:
Could we take the Ollama URL from configuration?

Reply:
I didn't mean forcing the dev to configure the URL themselves (e.g., using `dotnet user-secrets`). I just meant putting the URL into `appsettings.json` and using `Configuration` to read it. That doesn't cause any extra steps for the developer, but gives them more flexibility to change the value after deployment, for example.

Not a big deal and I don't really mind either way. It's easy for us to change at any time.